src/basic/process-util.c

```diff
@@ -1475,6 +1475,9 @@ int safe_fork_full(
         assert(!FLAGS_SET(flags, FORK_DETACH) ||
                (!ret_pid && (flags & (FORK_WAIT|FORK_DEATHSIG_SIGTERM|FORK_DEATHSIG_SIGINT|FORK_DEATHSIG_SIGKILL)) == 0));
 
+        if (FLAGS_SET(flags, FORK_TIOCSCTTY))
+                assert(FLAGS_SET(flags, FORK_SETSID));
```

Comment:
Is this really worth being shoved into safe_fork? Why not just introduce terminal_new_session() in terminal-util or so?

Reply:
I mean it's so simple we might as well just do it without a helper in that case, I don't mind, will get rid of the flags again

Reply:
i think safe_fork() is the better place (i suggested this after all), simply to make sure this is applied in the right order. 

(i'd not be totally opposed to a separate call, if people think, but that might give people the chance to call it too late or too early or so)

Reply:
Hmm, in what sense too late/early? safe_fork() always keeps the stdio fds around, so after safe_fork() it's definitely safe?

Reply:
too late/early relative to other shit people might be doing in the child after calling safe_fork(), that's all